src/Client.php

```diff
@@ -158,6 +171,72 @@ public function products(array $identities): Generator
         }
     }
 
+    /**
+     * @param string $sellerId
+     * @param ProductVersion $productVersion
+     * @param array<UuidInterface>|null $suppliersIds
+     * @param bool $disabled
+     */
+    public function createProduct(
+        string $sellerId,
+        ProductVersion $productVersion,
+        ?array $suppliersIds = null,
+        bool $disabled = false
+    ): Product {
+        $requestData = $productVersion->jsonSerialize();
+        $requestData['seller_id'] = $sellerId;
+        $requestData['disabled'] = $disabled;
+        if (null !== $suppliersIds) {
+            $requestData['suppliers'] = $suppliersIds;
+        }
+
+        $responseData = $this->sendRequest('post', 'product', $requestData);
+
+        return Product::fromData($responseData, $this->uriFactory);
+    }
+
+    /**
+     * @param string $sellerId
+     * @param ProductVersion $productVersion
+     * @param array<UuidInterface>|null $suppliersIds
+     * @param bool $disabled
+     */
+    public function updateProduct(
+        string $sellerId,
+        ProductVersion $productVersion,
+        ?array $suppliersIds = [],
+        bool $disabled = false
+    ): Product {
+        $requestData = $productVersion->jsonSerialize();
+        $requestData['disabled'] = $disabled;
+        if (null !== $suppliersIds) {
+            $requestData['suppliers'] = $suppliersIds;
+        }
+
+        $responseData = $this->sendRequest('put', "product/{$sellerId}", $requestData);
+
+        return Product::fromData($responseData, $this->uriFactory);
+    }
+
+    public function updateProductImage(string $sellerId, StreamInterface $imageStream, ?string $filename): void
+    {
+        $imageData = (string) $imageStream;
+        $mimeType = (new finfo())->buffer($imageData);
```

Comment:
As we already use `symfony/mime` we might use `MimeTypeGuesserInterface:: guessMimeType` and remove hard dependency on fileinfo